src/Cache/Engine/FileEngine.php

```diff
@@ -248,8 +248,8 @@ public function clear(): bool
             RecursiveIteratorIterator::SELF_FIRST
         );
         $cleared = [];
-        /** @var \SplFileInfo $fileInfo */
         foreach ($contents as $fileInfo) {
+            assert($fileInfo instanceof SplFileInfo);
```

Comment:
Doing the assert for each entry seems quite wasteful (even during development). This is how psalm itself deals with typing when iterating a recursive directory iterator https://github.com/vimeo/psalm/blob/b424de92d21df58d1ec62264550e20b6e39a3c85/src/Psalm/Config/FileFilter.php#L184-L185.